Add docs to consume preview packages (#544)
microsoft/cadl: README.md

If `cadl-server` cannot be found on PATH by Visual Studio in your setup, you can
configure its location by setting up the `cadl.cadl-server.path` entry in `.vs/VSWorkspaceSettings.json`. You may need to restart Visual Studio after changing this.
This should be the path to the `@cadl-lang/compiler` package. (e.g. `./node_modules/@cadl-lang/compiler`)

> ### Installing nightly version(@next tag)

Reviewer: ```suggestion
### Installing nightly version
```

Not convinced this adds much to the header.

On every commit to the main branch, packages with changes are automatically packaged and published under the `@next` tag under npm.

> In the [packages](#packages) section each package has a `@next` tag which show its current next version.

Reviewer: ```suggestion
The [packages](#packages) section shows which version corresponds to the `next` tag for each package.
```

> To use a `nightly` version of the packages update go over each one of the packages in the `package.json` fie and update it to either the latest published `@next` version or `@latest`, which ever is the newest. (You can also use the tag `latest` or `next`)

Reviewer 2: Are we saying that they can use the tag 'next' or 'latest' in their package.json files?  Or do we want them to be explicit about version and not use the tags?

Author: They can use either. For `next` I think it doesn't matter, `next` be maybe simpler. For `latest` we maybe should decide when we go to 1.0 if we want people to try to always use the latest version or be tied to a major, minor version

> To use a `nightly` version of the packages update go over each one of the packages in the `package.json` fie and update it to either the latest published `@next` version or `@latest`, which ever is the newest. (You can also use the tag `latest` or `next`)

Reviewer 2: ```suggestion
To use a `nightly` version of the packages, go over each one of the packages in the `package.json` fie and update it to either the latest published `@next` version or `@latest`, whichever is the newest. (You can also use the tag `latest` or `next` instead of an explicit version)
```

> After updating the packages you can run `npm update --force`. Force is required as there might be some incompatible version requirement.

Reviewer 2: ```suggestion
After updating the package.json file you can run `npm update --force`. Force is required as there might be some incompatible version requirement.
```
